components/dataHub/obs.c

```diff
@@ -1734,11 +1734,10 @@ dataSample_Ref_t obs_ApplyTransform
     dataSample_Ref_t sample = sampleRef;
     double transformVal;
 
-
     switch (obsPtr->transformType)
     {
         case OBS_TRANSFORM_TYPE_NONE:
-            break;
+            return sample;
```

Comment:
Is this a real problem that's being solved or are you just trying to suppress a compiler warning?  If the issue is that the compiler is warning that `transformVal` might be used uninitialized, then I think the better solution is just to initialize it to `0.0`.  I prefer that because it allows the function to retain a single return point.

Reply:
I think it was just a lint checker warning.

Reply:
Hmm...  I think there's a bug in the original code here.  I was just going to change that return to a goto and delete the condition at line 1764, but it looks like this function is destroying the sample that is stored in the observation's buffer.

Reply:
OK, just bad naming of UpdateSample().  It doesn't update the sample.  It creates a new one.